### PR TITLE
RavenDB-15044 - Add an overload for GetComapreExchangeValues which supports startsWith

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-15044

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Client/Documents/Session/ClusterTransactionSession.cs [hunk 1]
```diff
@@ -494,6 +523,8 @@ public interface IClusterTransactionOperations : IClusterTransactionOperationsBa
 
         Dictionary<string, CompareExchangeValue<T>> GetCompareExchangeValues<T>(string[] keys);
 
+        Dictionary<string, CompareExchangeValue<T>> GetCompareExchangeValues<T>(string startWith, int? start = null, int? pageSize = null);
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/ClusterTransactionSession.cs]: startWith => startsWith
int start = 0;
int pageSize = int.MaxValue

File: src/Raven.Client/Documents/Session/ClusterTransactionSession.cs [hunk 2]
```diff
@@ -514,6 +545,8 @@ public interface IClusterTransactionOperationsAsync : IClusterTransactionOperati
 
         Task<Dictionary<string, CompareExchangeValue<T>>> GetCompareExchangeValuesAsync<T>(string[] keys, CancellationToken token = default);
 
+        Task<Dictionary<string, CompareExchangeValue<T>>> GetCompareExchangeValuesAsync<T>(string startWith, int? start = null, int? pageSize = null, CancellationToken token = default);
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/ClusterTransactionSession.cs]: startWith => startsWith
int start = 0;
int pageSize = int.MaxValue

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/Documents/Session/ClusterTransactionSession.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/ClusterTransactionSession.cs]: Maybe
int start = 0,
int pageSize = 25

(like in `LoadStartingWith`)

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/ClusterTransactionSession.cs]: ok

File: src/Raven.Client/Documents/Session/ClusterTransactionSession.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/ClusterTransactionSession.cs]: done

File: src/Raven.Client/Documents/Session/ClusterTransactionSession.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/ClusterTransactionSession.cs]: done

**Review by ppekrol — APPROVED** [review]

**Pull request merged**